src/services/donations/updateEntityCounters.js

```diff
@@ -147,75 +148,80 @@ const updateEntity = async (app, id, type) => {
   }
 };
 
-const conversationSem = semaphore();
-
-const createConversation = async (context, donation, milestoneId) => {
+const createPaymentConversation = async (context, donation, milestoneId) => {
   const { app, method } = context;
   // Create payment conversation
   if (method === 'create' && donation.status === DonationStatus.PAID) {
-    app
-      .service('milestones')
-      .get(milestoneId)
-      .then(milestone => {
-        const { recipient } = milestone;
-        const { txHash, amount } = donation;
-        const { symbol, decimals } = donation.token;
-        const service = app.service('conversations');
-
-        conversationSem.take(async () => {
-          try {
-            const data = await service.find({
-              paginate: false,
-              query: {
-                milestoneId,
-                messageContext: 'payment',
-                txHash,
-                $limit: 1,
-              },
-            });
-
-            if (data.length > 0) {
-              const conversation = data[0];
-              const payments = conversation.payments || [];
-              const index = payments.findIndex(p => p.symbol === symbol);
+    const { txHash } = donation;
+    const events = (
+      await app.service('events').find({
+        query: {
+          transactionHash: donation.txHash,
+          event: 'Transfer',
+          status: { $nin: [EventStatus.PROCESSED, EventStatus.FAILED] },
```

Comment:
Did you test it? the event become processed after event is handled. So when the last event is handled, corresponding event would have processing state.... so the result length will not be zero at all and no email is sent!

Reply:
Yes I tested locally,  although for doesn't have accessing to email server the email not sent successfully but I printed the log, and the conversation created ( for one donation and two donation of different tokens I tested)

event became Processed and after that this function calls, so when reaching to this line if this event is the last one be sure that this query will return zero events

Reply:
@aminlatifi 

The reason why Donation was created after event status became `Processed` is not putting await before an async function, your guess was right,  Do you think we need to fix that problem during this issue (I vote for yes)?
https://github.com/Giveth/feathers-giveth/blob/d64f82c383a133c96c8b69874e7da06140725409/src/blockchain/pledges.js#L460-L465